test/controllers/CheckCsvFilesControllerSpec.scala

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.auth.RequestWithOptionalEmpRef
+import helpers.ErsTestHelper
+import models.CsvFiles
+import models.upscan.{UploadId, UploadedSuccessfully, UpscanCsvFilesList, UpscanIds}
+import org.mockito.ArgumentMatchers.{any, refEq}
+import org.mockito.Mockito.when
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Play.materializer
+import play.api.data.Form
+import play.api.http.Status
+import play.api.i18n
+import play.api.i18n.{Messages, MessagesImpl}
+import play.api.mvc.{AnyContent, DefaultMessagesControllerComponents, Request, Result}
+import play.api.test.FakeRequest
+import uk.gov.hmrc.http.cache.client.CacheMap
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration.Duration
+
+class CheckCsvFilesControllerSpec extends UnitSpec with GuiceOneAppPerSuite with ErsTestHelper {
+
+  lazy val mcc: DefaultMessagesControllerComponents = testMCC(fakeApplication())
+  implicit lazy val testMessages: MessagesImpl = MessagesImpl(i18n.Lang("en"), mcc.messagesApi)
+
+  def buildFakeCheckingServiceController(): CheckingServiceController =
+    new CheckingServiceController(mockAuthAction, mockUpscanService, mockSessionService, mcc, mockErsUtil, mockAppConfig) {
+      mockAnyContentAction
+    }
+
+  "selectCsvFilesPage" should {
+
+    "call showCheckCsvFilesPage if user is authenticated" in {
+      val controllerUnderTest = new CheckCsvFilesController(mockAuthAction, mcc, mockErsUtil, mockAppConfig) {
+        override def showCheckCsvFilesPage()(
+          implicit request: RequestWithOptionalEmpRef[AnyContent], hc: HeaderCarrier): Future[Result] = Future.successful(ImATeapot("here"))
```

Comment:
Not sure we needed to use the Teapot status code here?

Reply:
that's only there for ease of unit testing (if a 'teapot' status was thrown, it makes it perfectly clear it was mocked') - happy to replace it with something else though

Reply:
I see what you were going for, but I feel it can cause some confusion since the test doesn't reflect the code. Could be one to raise with the team though.